src/python/WMComponent/WorkQueueManager/WorkQueueManagerCleaner.py

```diff
@@ -47,9 +53,14 @@ def algorithm(self, parameters):
             # this will clean up whatever left over from above clean up.
             # also if the wq replication has problem it won't delay the killing jobs in condor
             # and updating wmbs status
-            abortedAndForceCompleteRequests = self.abortedAndForceCompleteWorkflowCache.getData()
+            # state lists which shouldn't be populated in wmbs. (To prevent creating work before WQE status updated)
+            # added completed list due to the race condition (The extra cost of the query is ~1sec per request)
+            requests = self.reqmgr2Svc.getRequestByStatusFromMemoryCache(["aborted", "aborted-completed", "force-complete", "completed"]).getData()
+            results = self.finishedWorflowCheck.execute(workflowNames=requests)
+
+            requestsToKill = [reqInfo["workflow"] for reqInfo in results if reqInfo["open"] > 0]
 
-            for wf in abortedAndForceCompleteRequests:
+            for wf in requestsToKill:
                 self.queue.killWMBSWorkflow(wf)
```

Comment:
Did you check whether it also kills logcollect and cleanup subscriptions/jobs?

Reply:
Yes, it doesn't kill LogCollect or Cleanup Jobs. Although as I mentioned killed job is not reported in wmstats but that might be different issue.